### PR TITLE
feat: Add /sendFile endpoint

### PR DESCRIPTION
**Pull request by lipis**
Produces this:

Screenshot 2020-03-31 at 1 16 12 PM

But we need this:

Screenshot 2020-03-31 at 1 16 28 PM

### CODE REVIEW

File: src/instance/InstanceFileOptions.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  @IsUUID('4')
+  conversationId!: string;
+
+  @ApiProperty()
+  @ValidateNested()
+  audio!: AudioMetaData;
+
+  @ApiProperty()
+  @ValidateNested()
+  video!: VideoMetaData;
```
Comment (lipis) [inline src/instance/InstanceFileOptions.ts]: how to expand those?

**Comment by lipis** [comment]
getting a validation error with the nested properties: 

```json
{
  "statusCode": 400,
  "message": [
    "audio.property durationInMillis should not exist",
    "audio.property normalizedLoudness should not exist",
    "video.property durationInMillis should not exist",
    "video.property height should not exist",
    "video.property width should not exist"
  ],
  "error": "Bad Request"
}
```

File: src/instance/InstanceController.ts [hunk 2]
```diff
@@ -491,6 +492,65 @@ export class InstanceController {
     }
   }
 
+  @Post(':instanceId/sendFile')
+  @ApiOperation({summary: 'Send an file to a conversation.'})
```
Comment (ffflorian) [inline src/instance/InstanceController.ts]:
```suggestion
  @ApiOperation({summary: 'Send a file to a conversation.'})
```

File: src/instance/InstanceFileOptions.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
+import {ApiProperty} from '@nestjs/swagger';
+import {LegalHoldStatus} from '@wireapp/core/dist/conversation/content/';
+import {IsBoolean, IsEnum, IsNumber, IsOptional, IsString, IsUUID, ValidateNested} from 'class-validator';
+
+class AudioMeta {
+  @ApiProperty()
+  @IsNumber()
+  durationInMillis!: number;
```
Comment (ffflorian) [inline src/instance/InstanceFileOptions.ts]:
```suggestion
  @IsOptional
  durationInMillis?: number;
```

File: src/instance/InstanceFileOptions.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+  @ApiProperty()
+  @IsNumber({}, {each: true})
+  normalizedLoudness!: number[];
```
Comment (ffflorian) [inline src/instance/InstanceFileOptions.ts]:
```suggestion
  @IsOptional
  normalizedLoudness?: number[];
```

File: src/instance/InstanceFileOptions.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+class VideoMeta {
+  @ApiProperty()
+  @IsNumber()
+  durationInMillis!: number;
```
Comment (ffflorian) [inline src/instance/InstanceFileOptions.ts]:
```suggestion
  @IsOptional
  durationInMillis?: number;
```

File: src/instance/InstanceFileOptions.ts [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+  @ApiProperty()
+  @IsNumber()
+  height!: number;
```
Comment (ffflorian) [inline src/instance/InstanceFileOptions.ts]:
```suggestion
  @IsOptional
  height?: number;
```

File: src/instance/InstanceFileOptions.ts [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+  @ApiProperty()
+  @IsNumber()
+  width!: number;
```
Comment (ffflorian) [inline src/instance/InstanceFileOptions.ts]:
```suggestion
  @IsOptional
  width?: number;
```

**Comment by ffflorian** [comment]
The audio and video properties are all optional, see `assetRoutes.ts`. :slightly_smiling_face:

**Pull request merged**